### ISSUE TITLE
use case-insensitive regexes for checking image extensions

### ISSUE THREAD

**Issue by likangning93** [AnalyticalGraphicsInc/gltf-pipeline#270]
See https://github.com/AnalyticalGraphicsInc/gltf-pipeline/blob/master/lib/loadGltfUris.js#L160

Instead of using a list of supported extensions and checking using `indexof`, we should use a regular expression when checking if the format is supported.

```
var ext = // some image format
var supportedExtensions = /\.(png|jpg|jpeg|bmp)$/i;
supportedExtensions.test(ext);
```

Thanks @mramato!

**Comment by HappyDash** [comment]
Another easy way to club jpeg and jpg together in regex.

var ext = // some image format
var supportedExtensions = /^.*\.(jpe?g|png|bmp)$/i;
supportedExtensions .test(ext);

**Comment by ryki2658** [comment]
I would like to help with this. It will be my first contribution to a open source project. I have forked and cloned the repository, I have made the changes that have been requested (...at least I have attempted to). Do you want me to now add commit and push the changes?

**Comment by lilleyse** [comment]
@ryki2658 Yes go ahead and then open a pull request.

Also since this is your first contribution you will need to sign a CLA before we merge any code: https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#contributor-license-agreement-cla

**Comment by ryki2658** [comment]
Ok I sent the agreement over. Is the pull request info similar to the commit comments?

**Comment by emackey** [comment]
@HappyDash @ryki2658 Would the regex behave any differently if you dropped `^.*` from the front?  Why explicitly specify "starts with... anything" ?

**Comment by HappyDash** [comment]
@emackey 
extensions start with dot So I thought this regex will clearly specify that required string should always start with dot

**Comment by mramato** [comment]
Long story short: `/\.(png|jpe?g|bmp)$/i` is the best option here.

@HappyDash your regex is actually incorrect because it does not verify the extension starts with a period.  See the below diagram ([source](https://regexper.com/#%2F%5E.*.(jpe%3Fg%7Cpng%7Cbmp)%24%2Fi)): 
![image](https://cloud.githubusercontent.com/assets/1495196/25806547/8d3a8e52-33d1-11e7-8cab-cbe3a046060b.png) 
You need to escape the period in order to have the regex check for it.

Contrast that to `/\.(png|jpg|jpeg|bmp)$/i;` which produces a simpler workflow:
![image](https://cloud.githubusercontent.com/assets/1495196/25806626/ce43aed8-33d1-11e7-9482-56612fe4ffc5.png)

So the original example is correct, but you are right in that it can be slightly shorter with `/\.(png|jpe?g|bmp)$/i` :
![image](https://cloud.githubusercontent.com/assets/1495196/25806684/0d6f00b2-33d2-11e7-8c64-dde9f6f2c507.png)

**Comment by emackey** [comment]
@mramato The regex was fixed in 7ab14d94b20dbf5cf27 in #278.

**Comment by HappyDash** [comment]
Oh yeah I am really sorry, I didn't check it properly. Actually I have just
learned writing some regexes at office. I didn't see it through. Thanks,
for explaining it properly.
Also, diagrams are so cool.
